crates/cast/bin/cmd/wallet/mod.rs

```diff
@@ -320,40 +345,123 @@ flag to set your key via:
+                        }
+                    }
+                }
+
+                // List ledger accounts
+                if ledger || all {
+                    match Ledger::new(LedgerHDPath::LedgerLive(0), 0).await {
+                        Ok(ledger) => {
+                            let mut live_addrs = String::new();
+                            let mut legacy_addrs = String::new();
+                            // get first 2 addresses from live and legacy derivation paths
+                            for i in 0..2 {
+                                if let Ok(address) =
+                                    ledger.get_address_with_path(&LedgerHDPath::LedgerLive(i)).await
+                                {
+                                    live_addrs.push_str(&format!("  -{}\n", address.to_alloy()));
+                                }
+
+                                if let Ok(address) =
+                                    ledger.get_address_with_path(&LedgerHDPath::Legacy(i)).await
+                                {
+                                    legacy_addrs.push_str(&format!("  -{}\n", address.to_alloy()));
                                 }
                             }
+                            println!(
+                                "Ledger\n LedgerLive HD path\n{} ...\n Legacy HD path\n{} ...",
+                                live_addrs, legacy_addrs
+                            );
+                        }
+                        Err(e) => {
+                            println!("{}", e)
                         }
                     }
-                    Err(e) => return Err(e),
+                }
+
+                // List Trezor accounts
+                if trezor || all {
+                    match Trezor::new(TrezorHDPath::TrezorLive(0), 0, None).await {
+                        Ok(trezor) => {
+                            let mut addrs = String::new();
+                            // get first 2 addresses from live derivation path
+                            for i in 0..2 {
+                                if let Ok(address) =
+                                    trezor.get_address_with_path(&TrezorHDPath::TrezorLive(i)).await
+                                {
+                                    addrs.push_str(&format!(" -{}\n", address.to_alloy()));
+                                }
+                            }
+                            println!("Trezor\n{} ...", addrs);
+                        }
+                        Err(e) => {
+                            println!("{}", e)
+                        }
+                    }
+                }
+
+                // List AWS account
+                if aws || all {
+                    let key_ids = std::env::var("AWS_KMS_KEY_IDS")
+                        .unwrap_or(std::env::var("AWS_KMS_KEY_ID").unwrap_or_default());
+
+                    if key_ids.is_empty() {
+                        println!("AWS KMS keys not configured");
```

Comment:
this should probably only print if not all?

Reply:
added in latest commit